lib/at_cmd_parser/at_cmd_parser.c

```diff
@@ -343,3 +340,29 @@ int at_parser_max_params_from_str(const char *at_params_str,
 
 	return err;
 }
+
+enum at_cmd_type at_parser_cmd_type_get(const char *at_cmd)
+{
+	enum at_cmd_type type;
+
+	if (!is_command(at_cmd)) {
+		return AT_CMD_TYPE_UNKNOWN;
+	}
+
+	skip_command_prefix(&at_cmd);
+
+	while (is_valid_notification_char(*at_cmd)) {
+		at_cmd++;
+	}
+
+	if ((*at_cmd == AT_CMD_SEPARATOR) &&
+	    (*(at_cmd + 1) == AT_CMD_READ_TEST_IDENTIFIER)) {
+		type = AT_CMD_TYPE_TEST_COMMAND;
+	} else if (*at_cmd == AT_CMD_READ_TEST_IDENTIFIER) {
+		type = AT_CMD_TYPE_READ_COMMAND;
+	} else {
+		type = AT_CMD_TYPE_SET_COMMAND;
```

Comment:
This "else" could lead to possible error. For example AT+COPS>0 instead of AT+COPS=0, i.e. any character not  AT_CMD_SEPARATOR.

> if (*at_cmd == AT_CMD_SEPARATOR) {
> 	   if  (*(at_cmd + 1) == AT_CMD_READ_TEST_IDENTIFIER) {
> 		type = AT_CMD_TYPE_TEST_COMMAND;
>               } else {
> 		type = AT_CMD_TYPE_SET_COMMAND;
>              }
> 	} else if (*at_cmd == AT_CMD_READ_TEST_IDENTIFIER) {
> 		type = AT_CMD_TYPE_READ_COMMAND;
> 	} else {
> 		type = AT_CMD_TYPE_UNKNOWN;
>     }

Reply:
Ok, I'll rework the `else` condition to be more specific here.

Reply:
Updated, I've kept single-level indentation but added a specific check for `=`/LF/CR/NULL. Otherwise set command type to unknown.